### ISSUE TITLE
Training Time Takes too Long (BiMPM Model)

### ISSUE THREAD

**Issue by tim5go** [nelson-liu/paraphrase-id-tensorflow#61]
Even I enable GPU computing (Nvidia tesla k80 X4) , it takes around 10 hours to finish 1 epoch.
Just wonder if this behavior is expected or not.

**Comment by nelson-liu** [comment]
I haven't implemented multiGPU yet, so 4xK80 is as slow as 1xK80. On a 1080 Ti, it takes 2h 18 mins per epoch, so 10 hours seems slow. What hyperparameters are you using?

**Comment by nelson-liu** [comment]
the model would be a lot faster with batch padding (tracked in https://github.com/nelson-liu/paraphrase-id-tensorflow/issues/44)

**Comment by tim5go** [comment]
I am using the default one you provided, i.e.
python scripts/run_model/run_bimpm.py train --early_stopping_patience=5 --model_name=biMPM --run_id=0

**Comment by tim5go** [comment]
Also, it seems that there's opening issue in the module "overrides"
See: mkorpela/overrides#11
So running in Python 3.6 env without modification will be a problem ^^"

**Comment by nelson-liu** [comment]
Hmm yes the ~2 hr timing j got was with those settings. Can you verify that you've uninstalled tensorflow and installed tensorflow-gpu (sorry if this seems obvious, just want to make sure).

Also, when training does nvidia-smi show that the gpu is being utilized?

**Comment by tim5go** [comment]
You are right, only one GPU is being used
And the Volatile GPU-Util is above 90% 
(So it means fully utilized, right?)

**Comment by nelson-liu** [comment]
Yes, that's what I wanted to check. I'm surprised that the K80 is that much slower, but looking online benchmarks do seem to show that 4x factor is about right. 

So yes those times are unfortunately expected for a K80 :( a good fix would be to add multigpu to enable bigger batch sizes, but I'm pretty low-bandwidth this week and won't have time to get to it. I expect it shouldn't be too hard, though...

**Comment by tim5go** [comment]
Too bad for K80, anyways thanks for your help ~
